document/value_test.go

```diff
@@ -21,6 +21,7 @@ func TestValueString(t *testing.T) {
 		{"bool", document.NewBoolValue(true), "true"},
 		{"int", document.NewIntegerValue(10), "10"},
 		{"double", document.NewDoubleValue(10.1), "10.1"},
+		{"double with trailing .0", document.NewDoubleValue(9223372036854775808), "9223372036854775808.0"},
```

Comment:
Can we also add a test case for `document.NewDoubleValue(1e24)`? Ideally, this should be `1e+24` (note that JSON integers may not have an exponent). See https://play.golang.org/p/AG47LJ-WEE_0

Reply:
In particular, this would cover `abs ≥ 1e21` branch where `fmt = 'e'`.

Reply:
Currently FormatFloat returns `1.0e+24` (that’s still a valid JSON number though).
See https://play.golang.org/p/0qpbJPaeoG_G

Reply:
Actually, `1e21` would be even better since it’s the boundary for exponent format.

Reply:
Yep good call. I think that it makes sense to have the trailing `.0` even in the `e` notation.